engine/src/main/java/org/terasology/persistence/serializers/GsonSerializer.java

```diff
@@ -52,47 +71,112 @@ public GsonSerializer() {
+     * @param object the file to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param file file that the bytes will be written to
+     * @throws IOException gets thrown if there is an issue writing to the file
+     */
     public <T> void writeJson(T object, TypeHandler<T> typeHandler, File file) throws IOException {
         try (Writer writer = new BufferedWriter(new FileWriter(file))) {
             writeJson(object, typeHandler, writer);
         }
     }
 
-    public <T> void writeJson(T object, TypeHandler<T> typeHandler, String path) throws IOException {
-        writeJson(object, typeHandler, new File(path));
+    /**
+     * Writes the a the object's persisted data to a File located at a specified file name
+     * 
+     * @see #writeJson(Object, TypeHandler, String)
+     * @param object the object to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param fileName the file name where the JSON will be written
+     * @throws IOException gets thrown if there is an error writing to the file at the specified location
+     */
+    public <T> void writeJson(T object, TypeHandler<T> typeHandler, String fileName) throws IOException {
+        writeJson(object, typeHandler, new File(fileName));
     }
 
+    /**
+     * Gets the PersistedData from a Reader object
+     * 
+     * @see #persistedDataFromJson(InputStream)
+     * @param reader Reader object that will be deserialized
+     * @return deserialized GsonPersistedData object
+     */
     public PersistedData persistedDataFromJson(Reader reader) {
         JsonElement jsonElement = gson.fromJson(reader, JsonElement.class);
 
         return new GsonPersistedData(jsonElement);
     }
 
+    /**
+     * Gets the PersistedData from an InputStream
+     * 
+     * @see #persistedDataFromJson(Reader)
+     * @param stream the InputStream that will be serialized
+     * @return deserialized GsonPersistedData object
+     * @throws IOException if there is an issue parsing the stream
+     */
     public PersistedData persistedDataFromJson(InputStream stream) throws IOException {
         try (Reader reader = new InputStreamReader(stream)) {
             return persistedDataFromJson(reader);
         }
     }
 
+    /**
+     * Gets the PersistedData from a File object
+     * 
+     * @see #persistedDataFromJson(String)
+     * @param file File object containing the JSON that will be deserialized
+     * @return deserialized GsonPersistedData object
+     * @throws IOException gets thrown if there is an issue reading the File object
+     */
     public PersistedData persistedDataFromJson(File file) throws IOException {
         try (Reader reader = new FileReader(file)) {
             return persistedDataFromJson(reader);
         }
     }
 
+    /**
+     * Gets the PersistedData by forwarding a Reader which contains a String to {@link #persistedDataFromJson(Reader)}
```

Comment:
Like above, please add more details. Also, the implementation specifics (`by forwarding a Reader which contains a String to {@link #persistedDataFromJson(Reader)}`) are irrelevant here, what matters more is what the method does on a more abstract level.